docs/docs/spec.md

```diff
@@ -927,7 +927,7 @@ the output of the following operation:
     > demonstrated in
     > https://github.com/jspecify/jspecify-reference-checker/pull/197.
 
-    > The purpose of this special case is to improve behavior in "the
+    > The primary purpose of this special case is to improve behavior in "the
     > `ImmutableList.Builder` case": Because `ImmutableList.Builder.add` always
```

Comment:
I'd love to avoid referring to a Guava class. Could we motivate this with a hypothetical class (or one in the JDK&mdash;what about `Comparable`?), to make it clear that this isn't really a special case? It's asymmetrical, but that's a function of the fact that `Foo!` is a strict subtype of `Foo?`.

But if we're going to refer to `ImmutableList.Builder`, shouldn't we at least link to it? It's not the case that everyone reading this spec will be intimately familiar with that API.

Reply:
I made an attempt to rewrite this to be about `Comparable` and to make clear that it's not a special case. PTAL.